test/extended/networking/bonds.go

```diff
@@ -0,0 +1,117 @@
+		namespace := f.Namespace.Name
+		podName1, podName2 := "pod1", "pod2"
+		bondnad1, bondnad2 := "bondnad1", "bondnad2"
+
+		g.By("creating network attachment definitions")
+		o.Expect(
+			createNetworkAttachmentDefinition(
+				oc.AdminConfig(),
+				namespace,
+				"macvlannad",
+				`{"cniVersion":"0.4.0","name":"macvlan-nad","plugins":[{ "type": "macvlan"}]}`),
+		).To(
+			o.Succeed(),
+			"unable to create macvlan network-attachment-definition",
+		)
+
+		o.Expect(
+			createBondNAD(
+				oc.AdminConfig(),
+				namespace,
+				bondnad1,
+				"10.0.0.10/24",
+				"net1",
+				"net2",
+			),
+		).To(
+			o.Succeed(),
+			"unable to create bond network-attachment-definition",
+		)
+
+		o.Expect(
+			createBondNAD(
+				oc.AdminConfig(),
+				namespace,
+				bondnad2,
+				"10.0.0.11/24",
+				"net1",
+				"net2",
+			),
+		).To(
+			o.Succeed(),
+			"unable to create bond network-attachment-definition",
+		)
+
+		g.By("creating first pod and checking results")
+		exutil.CreateExecPodOrFail(f.ClientSet, namespace, podName1, func(pod *kapiv1.Pod) {
+			macVlanAnnotation := fmt.Sprintf("%s/%s", namespace, "macvlannad")
+			bondAnnotation := fmt.Sprintf("%s/%s@%s", namespace, bondnad1, bondnad1)
+			pod.ObjectMeta.Annotations = map[string]string{"k8s.v1.cni.cncf.io/networks": fmt.Sprintf("%s, %s, %s", macVlanAnnotation, macVlanAnnotation, bondAnnotation)}
+		})
+		pod, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(context.TODO(), podName1, metav1.GetOptions{})
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		networkStatusString, ok := pod.Annotations["k8s.v1.cni.cncf.io/network-status"]
+		o.Expect(ok).To(o.BeTrue())
+		o.Expect(networkStatusString).ToNot(o.BeNil())
+		networkStatuses := []nadtypes.NetworkStatus{}
+		o.Expect(json.Unmarshal([]byte(networkStatusString), &networkStatuses)).ToNot(o.HaveOccurred())
+		o.Expect(networkStatuses).To(o.HaveLen(4))
+		o.Expect(networkStatuses[3].Interface).To(o.Equal(bondnad1))
+		o.Expect(networkStatuses[3].Name).To(o.Equal(fmt.Sprintf("%s/%s", namespace, bondnad1)))
+
+		g.By("having a second pod pinging the first pod")
+		exutil.CreateExecPodOrFail(f.ClientSet, namespace, podName2, func(pod *kapiv1.Pod) {
+			macVlanAnnotation := fmt.Sprintf("%s/%s", namespace, "macvlannad")
+			bondAnnotation := fmt.Sprintf("%s/%s@%s", namespace, bondnad2, bondnad2)
+			pod.ObjectMeta.Annotations = map[string]string{"k8s.v1.cni.cncf.io/networks": fmt.Sprintf("%s, %s, %s", macVlanAnnotation, macVlanAnnotation, bondAnnotation)}
+			pod.Spec.Containers[0].Command = []string{"/bin/bash", "-c", fmt.Sprintf("ping -c 1 %s", "10.0.0.10")}
+		})
+	})
+})
+
+func createBondNAD(config *rest.Config, namespace string, nadName string, ip string, slaveNames ...string) error {
+	slaves := []string{}
+	for _, name := range slaveNames {
+		slaves = append(slaves, fmt.Sprintf("{\"name\": \"%s\"}", name))
+	}
+	links := strings.Join(slaves, ",")
+	nadConfig := fmt.Sprintf(`{
+      "type": "bond",
```

Comment:
indentation here is weird... it would be better to indent the contents to beyond the `nadConfig :=` line. Using `"github.com/lithammer/dedent"` might help here

Reply:
Done